pulpcore/cli/common/generic.py

```diff
@@ -66,31 +132,6 @@ def float_or_empty(value: str) -> Union[str, float]:
 # Custom classes for commands and parameters
 
 
-class ClickNoWait(click.ClickException):
-    exit_code = 0
-
-    def show(self, file: Optional[IO[str]] = None) -> None:
-        """
-        Format the message into file or STDERR.
-        Overwritten from base class to not print "Error: ".
-        """
-        if file is None:
-            file = click.get_text_stream("stderr")
-        click.echo(self.format_message(), file=file)
-
-
-class PulpCLIContext(PulpContext):
-    """
-    Subclass of the Context that overwrites the CLI specifics.
-    """
-
-    def echo(self, message: str, nl: bool = True, err: bool = False) -> None:
-        click.echo(message, nl=nl, err=err)
-
-    def prompt(self, text: str, hide_input: bool = False) -> Any:
-        return click.prompt(text, hide_input=hide_input)
```

Comment:
How was this working before? `pass_pulp_context` was making a decorator from `PulpContext`, how was any command able to output to the terminal?

Reply:
`pass_pulp_context` would return the first found **subclass of `PulpContext`** context object in the context hierarchy. This subclass happened to be the cli specific type.